backends-velox/src/main/scala/org/apache/gluten/execution/FilterExecTransformer.scala

```diff
@@ -22,13 +22,22 @@ import org.apache.spark.sql.execution.SparkPlan
 case class FilterExecTransformer(condition: Expression, child: SparkPlan)
   extends FilterExecTransformerBase(condition, child) {
   override protected def getRemainingCondition: Expression = {
-    val scanFilters = child match {
-      // Get the filters including the manually pushed down ones.
-      case basicScanExecTransformer: BasicScanExecTransformer =>
-        basicScanExecTransformer.filterExprs()
-      // For fallback scan, we need to keep original filter.
-      case _ =>
-        Seq.empty[Expression]
+    var scanFilters: Seq[Expression] = Seq()
```

Comment:
Would you use match case instead of using `isInstanceOf`?

Reply:
`IcebergScanTransformer`/`DeltaScanTransformer` is inherited from abstract class and can't be matched using case.

Reply:
An abstract class can be used in `match-case`.

Reply:
Can this not work?
```scala
val scanFilters = child match {
      case base: BatchScanExecTransformerBase =>
        base.filterExprs();
      case base: FileSourceScanExecTransformerBase =>
        base.filterExprs();
      case _ =>
        child match {
          // Get the filters including the manually pushed down ones.
          case basicScanExecTransformer: BasicScanExecTransformer =>
            basicScanExecTransformer.filterExprs()
          // For fallback scan, we need to keep original filter.
          //      case batchScanExecTransformer: BatchScanExecTransformerBase =>
          //        batchScanExecTransformer.filterExprs()
          case _ =>
            Seq.empty[Expression]
        }
    }
```